pkg/agent/informant.go

```diff
@@ -874,6 +908,50 @@ func (s *InformantServer) handleTryUpscale(
 	}
 }
 
+// HealthCheck makes a request to the informant's /health-check endpoint, using the server's ID.
+//
+// This method MUST be called while holding i.server.requestLock AND NOT i.server.runner.lock.
+func (s *InformantServer) HealthCheck(ctx context.Context) (*api.InformantHealthCheckResp, error) {
+	s.requestLock.Lock()
+	defer s.requestLock.Unlock()
+
+	err := func() error {
+		s.runner.lock.Lock()
+		defer s.runner.lock.Unlock()
+		return s.Valid()
```

Comment:
The reason it can't hold the lock is because otherwise it's impossible to know if the thing you currently have is valid — the only way to not have racy behavior is to do the "is it valid" check while you're holding a lock beyond it

Reply:
I guess it's confusing from a lock hygiene perspective: the `s` method contains the locks in needs to return, but it can't take them directly?

I think based on this explanation, `Valid` should be private because it's not possible to use it (correctly/safely) outside of the `agent` package.

Reply:
That makes sense. Opened #215.